mountpoint-s3/src/inode.rs

```diff
@@ -131,7 +131,7 @@ impl Superblock {
     /// The kernel may forget a number of references (`n`) in one forget message to our FUSE implementation.
     /// If the lookup count reaches zero, it is safe for the [Superblock] to delete the [Inode].
     pub fn forget(&self, ino: InodeNo, n: u64) {
-        let mut inodes = self.inner.inodes.write().unwrap();
+        let inodes = self.inner.inodes.read().unwrap();
         match inodes.get(&ino) {
```

Comment:
Sorry for churn, but would it be safest/clearest to just have:

```suggestion
        let inode = {
            let inodes = self.inner.inodes.read().unwrap();
            inodes.get(&ino).cloned()
        }
        match inode {
```

This would avoid the need to drop `inodes` explicitly later.

We can even pull the `None` case into this scope and then just return an unwrapped non-option inode.